### DISCUSSION TITLE
Change language in app

### DISCUSSION THREAD

**Discussion by cuongnv-dev** [infinitered/ignite#1732] [Q&A] [answered]
I want to apply change language feature in app (Not using default Localization). I just save the lang code in storage,  So in the i18n.ts, how can I change language when lang code change? 
Hope get your help

**Comment by robinheinze** [comment, marked as answer]
Good question! Ignite comes with [expo-localization](https://docs.expo.io/versions/latest/sdk/localization), which is a great starter library, but is somewhat limited and only supports the user's default device locale, so changing the language means changing device settings. If you're looking for a more robust localization solution that allows passing a locale value via code, [React i18n-next](https://react.i18next.com/) is a great option!
